File: easy/922. Sort Array By Parity II/solution.java
class Solution {
    public int[] sortArrayByParityII(int[] nums) {
        int even = 0;
        int odd = 1;
        
        while (even < nums.length && odd < nums.length) {
            if (nums[even] % 2 == 0) {
                even += 2;
            } else {
                int temp = nums[even];
                nums[even] = nums[odd];
                nums[odd] = temp;
                odd += 2;
            }
        }
        
        return nums;
    }
}

File: easy/922. Sort Array By Parity II/solution.rs
impl Solution {
    pub fn sort_array_by_parity_ii(mut nums: Vec<i32>) -> Vec<i32> {
        let mut even = 0;
        let mut odd = 1;
        
        while even < nums.len() && odd < nums.len() {
            if nums[even] % 2 == 0 {
                even += 2;
            } else {
                nums.swap(even, odd);
                odd += 2;
            }
        }
        
        nums
    }
}